### PR TITLE
Post management pull request

### PR DESCRIPTION
**Pull request by gareth-go**
# Description
Import select2
Feature: create, show post

# Screenshot
image
image

# Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/posts/new.html.slim
Comment (gary-gia-goldenowl) [inline app/views/posts/new.html.slim:1]: Should not use inline css, just in some specific case, must separate css file for single module to easy maintain (for example module user have user.scss file, ....)

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  get 'posts/new', to: 'posts#new', as: :new_post
+  get 'post/:id', to: 'posts#show', as: :post
+  post 'posts', to: 'posts#create', as: :posts
```
Comment (gary-gia-goldenowl) [inline config/routes.rb]: Should use **resources** to make them more concise instead of separate actions

File: app/javascript/packs/create-post.js
Comment (gary-gia-goldenowl) [inline app/javascript/packs/create-post.js:1]: - Ruby and Rails use snake_case for naming conventions, where words are separated by underscores, so i suggest using underscore convention for SCSS and JavaScript files to make them consistent with the way Ruby on Rails files (https://gist.github.com/iangreenleaf/b206d09c587e8fc6399e?permalink_comment_id=3291230)

File: app/javascript/stylesheets/create-post.scss
Comment (gary-gia-goldenowl) [inline app/javascript/stylesheets/create-post.scss:1]: - Ruby and Rails use snake_case for naming conventions, where words are separated by underscores, so i suggest using underscore convention for SCSS and JavaScript files to make them consistent with the way Ruby on Rails files

- The relative path should separate scss and js file for example:  **/app/javascript/scss/create_post.scss** (https://guides.rubyonrails.org/webpacker.html#using-webpacker-for-css)

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+  # get 'posts/new', to: 'posts#new', as: :new_post
+  # get 'post/:id', to: 'posts#show', as: :post
+  # post 'posts', to: 'posts#create', as: :posts
+  resources :posts
```
Comment (nathan-phan-goldenowl) [inline config/routes.rb]:
```
resources :posts, only: %i[show create new]
```

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      render json: @post.errors.messages
+    end
+  end
+
+  private
+
+  def set_new_post
+    # set status of new post is accepted if user is admin
+    @post = Post.new(user_id: current_user.id)
+    @post.status = 1 if current_user.admin?
```
Comment (nathan-phan-goldenowl) [inline app/controllers/posts_controller.rb]:
```
post.status = :accepted
```

**Pull request merged**